src/Board.py

```diff
@@ -50,6 +52,79 @@ def __init__(self, style: str = "default", size: int = 5) -> None:
+        return self.data[coords[1]][coords[0]]
+
+    def get_directly_adj(
+        self,
+        start: Coords = (0, 0),
+        terrain_filter: Terrain | None = None,
+    ) -> List[Coords]:
+        adjacent_tile_list = []
+        row_start_offset = -((start[1] + 1) % 2)
+        if terrain_filter is None:
+            if start[1] > 0:
+                adjacent_tile_list.append(
+                    (start[0] + row_start_offset, start[1] - 1)
+                )  # top left
+                adjacent_tile_list.append(
+                    (start[0] + row_start_offset + 1, start[1] - 1)
+                )  # top right
+            if start[0] < BOARD_WIDTH - 2 or (
+                start[1] % 2 == 1 and start[0] < BOARD_WIDTH - 1
+            ):
+                adjacent_tile_list.append((start[0] + 1, start[1]))  # right
+            if start[1] < BOARD_HEIGHT - 1:
+                adjacent_tile_list.append(
+                    (start[0] + row_start_offset + 1, start[1] + 1)
+                )  # bottom left
+                adjacent_tile_list.append(
+                    (start[0] + row_start_offset, start[1] + 1)
+                )  # bottom right
+            if start[0] > 0:
+                adjacent_tile_list.append((start[0] - 1, start[1]))  # left
+        else:
+            unfiltered_list = self.get_directly_adj(start, None)
+            for tile in unfiltered_list:
+                if self.get(tile)._terrain == terrain_filter:
+                    adjacent_tile_list.append(tile)
+        return adjacent_tile_list  # = [TL,TR,R,BR,BL,L]
+
+    def get_indirectly_adj(
+        self, start=(0, 0), terrain_filter=None, shipping_limit=0
+    ) -> list:
+        unfiltered_list = []
+        if self.get_directly_adj(start, Terrain.RIVER) == []:
+            return self.get_directly_adj(start, terrain_filter)
+        if shipping_limit == 0:
+            unfiltered_list = self.get_directly_adj(start)
+        else:
+            direct_list = self.get_directly_adj(start, None)
+            for tile in direct_list:
+                if tile not in unfiltered_list:
+                    unfiltered_list.append(tile)
+                if self.get(tile).terrain == Terrain.RIVER:
+                    indirect_list = self.get_indirectly_adj(
+                        tile, None, shipping_limit - 1
+                    )
+                    for t in indirect_list:
+                        if t not in unfiltered_list and t != start:
+                            unfiltered_list.append(t)
+        if terrain_filter is None:
+            return unfiltered_list
+        adjacent_tile_list = []
+        for tile in unfiltered_list:
+            if self.get(tile)._terrain == terrain_filter:
+                adjacent_tile_list.append(tile)
+        return adjacent_tile_list
+
+    def check_adjacency(self, start, end, shipping_limit) -> str:
+        if end in self.get_directly_adj(start, None):
+            return "Direct"
```

Comment:
As a general note, instead of having "magic numbers" (or magic strings in this case) in the middle of code, it's better to have some kind of a value that can be type-checked and have other static analysis (i.e. analysis that doesn't involve running the code) run on it.

For example, this could use an Enum:
```python3
from enum import Enum

class AdjacencyType(Enum):
    DIRECT = 0
    INDIRECT = 1

...

return AdjacencyType.DIRECT
```